source/fundamentals/data-formats/document-data-format-pojo.txt

```diff
@@ -13,17 +13,226 @@ Document Data Format: POJOs
+   and on which classes to apply the Codecs. In this example, we use the
+   :java-docs:`automatic(true) <apidocs/bson/org/bson/codecs/pojo/PojoCodecProvider.Builder.html#automatic(boolean)>`
+   setting of the :java-docs:`PojoCodecProvider.Builder <apidocs/bson/org/bson/codecs/pojo/PojoCodecProvider.Builder.html>`
+   to apply the Codecs to any class and its properties.
+
+   .. code-block:: java
+
+      CodecProvider pojoCodecProvider = PojoCodecProvider.builder().automatic(true).build();
+
+.. note::
+
+   Codec providers also contain other objects such as ``ClassModel`` and
+   ``Convention`` instances that further define serialization behavior.
+   For more information on codec providers and customization, see our guide
+   on :doc:`POJO Customization </fundamentals/data-formats/pojo-customization>`.
+
+
+#. Add the ``PojoCodecProvider`` instance to a ``CodecRegistry``. The
+   ``CodecRegistry`` allows you to specify one or more codec providers to
+   encode the POJO data. In this example, we call the following methods:
+
+   - :java-docs:`fromRegistries() <apidocs/bson/org/bson/codecs/configuration/CodecRegistries.html#fromRegistries(org.bson.codecs.configuration.CodecRegistry...)>`
+     to combine multiple ``CodecRegistry`` instances into a single one
+   - :java-docs:`getDefaultCodecRegistry() <apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.html?is-external=true#getDefaultCodecRegistry()>`
+     to retrieve a ``CodecRegistry`` instance from a list of codec providers
+   - :java-docs:`fromProviders() <apidocs/bson/org/bson/codecs/configuration/CodecRegistries.html#fromProviders(org.bson.codecs.configuration.CodecProvider...)>`
+     to create a ``CodecRegistry`` instance from the ``PojoCodecProvider``
+
+   See the code below to see how to instantiate the ``CodecRegistry``:
+
+   .. code-block:: java
+
+      // ensure you use the following static imports above your class definition
+      import static com.mongodb.MongoClientSettings.getDefaultCodecRegistry;
+      import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+      import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+      // ...
+
+      CodecRegistry pojoCodecRegistry = fromRegistries(getDefaultCodecRegistry(), fromProviders(pojoCodecProvider));
+
+#. Configure our ``MongoClient``, ``MongoDatabase``, or ``MongoCollection``
+   instance to use the Codecs in the ``CodecRegistry``. You only need to
+   configure one of these. In this example, we set the ``CodecRegistry`` on a
+   ``MongoDatabase`` called ``sample_pojos`` using the ``withCodecRegistry()``
+   method.
+
+   .. code-block:: java
+
+      try (MongoClient mongoClient = MongoClients.create(uri)) {
+          MongoDatabase database = mongoClient.getDatabase("sample_pojos").withCodecRegistry(pojoCodecRegistry);
+          // ...
+      }
+
+#. Configure your ``MongoCollection`` instance to cast to your POJO class when
```

Comment:
See above note about casting

Reply:
Simplified the text a bit since the generic type was explained earlier, let me know what you think:

> Pass your POJO class to your call to ``getCollection()`` as the document class parameter and specify it as the type argument of your ``MongoCollection`` instance as follows: